pandas/core/frame.py

```diff
@@ -4144,6 +4144,13 @@ def _clear_item_cache(self) -> None:
 
     def _get_item_cache(self, item: Hashable) -> Series:
         """Return the cached item, item represents a label indexer."""
+        if (
+            get_option("mode.copy_on_write")
```

Comment:
Should we make a function for this?

Reply:
We already have one, but fixing it in https://github.com/pandas-dev/pandas/pull/50675. Will update this PR to use that function, or update the other PR to replace it, depending on what gets merged first.

Reply:
Updated to use the function in several places where we were checking this